### DISCUSSION TITLE
Enhance MethodCallToPropertyFetchRector to allow specifying a base class

### DISCUSSION THREAD

**Discussion by brandonkelly** [rectorphp/rector#7031] [Ideas]
I would like to find any code that is calling a certain method (`craft\elements\User::getFullName()`) and have them replaced with a property accessor (`$fullName`), for example:

```diff
- return $someUser->getFullName();
+ return $someUser->fullName;
```

It seems like [MethodCallToPropertyFetchRector](https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#methodcalltopropertyfetchrector) is close, but is only for `$this->methodName()`, and doesn’t allow specifying a base class for the method.

Hoping something like this could be made possible:

```php
$services->set(MethodCallToPropertyFetchRector::class)
    ->configure([
        new MethodCallToPropertyFetch('craft\elements\User', 'getFullName', 'fullName'),
    ]);
```

**Comment by TomasVotruba** [comment]
Resolved in https://github.com/rectorphp/rector-src/pull/1905
